php/context/class-fieldmanager-context-term.php

```diff
@@ -288,6 +298,40 @@ public function term_fields( $html_template, $taxonomy, $term = null ) {
 	 * @param string $taxonomy The term taxonomy.
 	 */
 	public function save_term_fields( $term_id, $tt_id, $taxonomy ) {
+		// phpcs:disable WordPress.Security.NonceVerification.Missing
+		// Ensure that the term being created/updated is the term in the request.
+		if ( ! empty( $_POST['tag_ID'] ) && $term_id !== (int) $_POST['tag_ID'] ) {
+			return;
+		} elseif (
+			empty( $_POST['tag_ID'] )
+			&& ! empty( $_POST['taxonomy'] )
+			&& $taxonomy === $_POST['taxonomy']
+			&& isset( $_POST['tag-name'], $_POST['parent'] )
+		) {
+			// Ensure this was the created_term action.
+			if ( ! doing_action( 'created_term' ) ) {
+				return;
+			}
+
+			if ( empty( $this->inserting_term_data ) ) {
+				// Something has gone awry, this shouldn't happen, so bail.
+				return;
+			}
+
+			// Core expects terms to have a unique combination of [taxonomy, name, parent].
+			$term = get_term( $term_id );
+			if ( $term->name !== $this->inserting_term_data['name'] ) {
```

Comment:
Will this condition match terms whose names contain characters that are automatically converted to HTML entities on save? For example, a term named `This & That` will be saved as `This &amp; That` (via `sanitize_term() -> 'pre_term_name' -> _wp_specialchars()`).